### PR TITLE
Remove unsupported group_shard_failures parameter

### PR DESCRIPTION
**Pull request by javanna**
We have had support for the `group_shard_failures` parameter in our code for a while, since we introduced failures grouping. When we introduced validation of parameters at REST, we seem to have forgotten to expose such parameter. Given that the parameter is effectively not supported for many months now, that no user has complained about that and that grouping is the expected behaviour, this commit removes support for the parameter.

Relates to #32598

### CODE REVIEW

**Comment by javanna** [comment]
retest this please

File: server/src/main/java/org/elasticsearch/action/search/SearchPhaseExecutionException.java [hunk 1]
```diff
@@ -134,11 +134,10 @@ private static String buildMessage(String phaseName, String msg, ShardSearchFail
     @Override
     protected void metadataToXContent(XContentBuilder builder, Params params) throws IOException {
         builder.field("phase", phaseName);
-        final boolean group = params.paramAsBoolean("group_shard_failures", true); // we group by default
-        builder.field("grouped", group); // notify that it's grouped
+        builder.field("grouped", true); // notify that it's grouped
```
Comment (colings86) [inline server/src/main/java/org/elasticsearch/action/search/SearchPhaseExecutionException.java]: I assume this is only here for backwards compatibility?

**Review by colings86 — APPROVED** [review]
LGTM, left a small note but fine to merge

File: server/src/main/java/org/elasticsearch/action/search/SearchPhaseExecutionException.java (hunk 1 above)
Comment (javanna) [reply server/src/main/java/org/elasticsearch/action/search/SearchPhaseExecutionException.java]: yea I did not want to change the output format at this stage. Also, given that the flag was not exposed, we would always return true anyways even before the change. We should remove this at some point though.

Comment (colings86) [reply server/src/main/java/org/elasticsearch/action/search/SearchPhaseExecutionException.java]: 👍

**Pull request merged**